File: testdata/bench/kotlin/BenchmarkTest00006.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via string interpolation in variable
suspend fun handler00006(call: ApplicationCall) {
    val table = call.parameters["table"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val sql = "SELECT COUNT(*) FROM $table"
    conn.createStatement().executeQuery(sql)
    call.respondText("OK")
}

File: testdata/bench/go/BenchmarkTest00006.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via defer
func Handler00006(w http.ResponseWriter, r *http.Request) {
	param := r.PostFormValue("table")
	db, _ := sql.Open("sqlite3", ":memory:")
	query := "DROP TABLE " + param
	defer db.Exec(query)
}
